core/src/main/scala/chisel3/Data.scala

```diff
@@ -372,6 +372,89 @@ trait BaseType extends HasId with NamedComponent {
     */
   private[chisel3] def cloneTypeFull: this.type
 
+  // Internal API: returns a ref that can be assigned to, if consistent with the binding
+  private[chisel3] def lref: Node = {
+    requireIsHardware(this)
+    requireVisible()
+    topBindingOpt match {
+      case Some(binding: ReadOnlyBinding) =>
+        throwException(s"internal error: attempted to generate LHS ref to ReadOnlyBinding $binding")
+      case Some(ViewBinding(target)) => reify(target).lref
+      case Some(binding: TopBinding) => Node(this)
+      case opt => throwException(s"internal error: unknown binding $opt in generating LHS ref")
+    }
+  }
+
+  // Internal API: returns a ref, if bound
+  private[chisel3] final def ref: Arg = {
```

Comment:
There's a lot of code down below that is having to be specialized for `Data` which suggests to me that this isn't really all that useful for non-Data BaseTypes, what do you actually need from this on Properties? Should instead it just be a virtual method that both Data and Property implement?

Reply:
Thanks for pointing this out, I realized it is not actually necessary to define lref and ref on BaseType. These were moved over in this PR in an earlier version, when I was also making issueConnect work with BaseType. I will revert this part, and we can define them for Property separately.